rest-service/manager_rest/resource_manager.py

```diff
@@ -1663,74 +1662,30 @@ def create_tasks_graph(self, name, execution_id, operations=None):
-                 for n in self.sm.list(models.Node, filters=filters,
-                                       get_all_results=True)
-                 if self._is_agent_node(n)}
-
-        if not nodes:
-            pagination = {'total': 0, 'size': 0, 'offset': 0}
-            return ListResult([], metadata={'pagination': pagination})
-
-        instance_filters = {'_node_fk': list(nodes)}
-        if node_instance_ids:
-            instance_filters['id'] = node_instance_ids
-        node_instances = self.sm.list(models.NodeInstance,
-                                      filters=instance_filters,
-                                      get_all_results=True)
-        agents = []
-        for inst in node_instances:
-            agent = self._agent_from_instance(inst)
-            if agent is None:
-                continue
-            if install_method is not None and \
-                    agent['install_method'] not in install_method:
-                continue
-            agents.append(agent)
-
-        # there's no real way to implement pagination in a meaningful manner,
-        # since we need to query node-instances anyway; still, include the
-        # sizes to keep response object shape consistent with all other
-        # endpoints
-        pagination = {'total': len(agents), 'size': len(agents), 'offset': 0}
-        return ListResult(agents, metadata={'pagination': pagination})
-
-    def _agent_from_instance(self, instance):
-        if instance.state != 'started':
-            return
-        agent = instance.runtime_properties.get('cloudify_agent')
-        if not agent:
-            return
-        if agent.get('windows'):
-            system = 'windows'
-        else:
-            system = agent.get('distro')
-            if agent.get('distro_codename'):
-                system = '{0} {1}'.format(system, agent.get('distro_codename'))
-        return dict(
-            id=instance.id,
-            host_id=instance.host_id,
-            ip=agent.get('ip'),
-            install_method=agent.get('install_method'),
-            system=system,
-            version=agent.get('version'),
-            node=instance.node_id,
-            deployment=instance.deployment_id
-        )
-
-    def _is_agent_node(self, node):
-        if cloudify_constants.COMPUTE_NODE_TYPE not in node.type_hierarchy:
-            return False
-        if cloudify_utils.internal.get_install_method(node.properties) == \
-                cloudify_constants.AGENT_INSTALL_METHOD_NONE:
-            return False
-        return True
+            filters['node_id'] = node_ids
+        if node_instance_ids is not None:
+            filters['node_instance_id'] = node_instance_ids
+        if install_method is not None:
+            filters['install_method'] = install_method
+        agents_raw = self.sm.list(models.Agent,
+                                  filters=filters,
+                                  all_tenants=all_tenants)
+        agents = [dict(id=agent.id,
```

Comment:
any reason to create a new list of dicts rather than just returning agents_raw directly?